modem-manager-gui: 0.0.19.1 -> 0.0.20

Galagora opened the pull request:
<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!
List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://hydra.nixos.org/job/nixpkgs/trunk/manual/latest/download/1/nixpkgs/manual.html#chap-reviewing-contributions
-->

###### Motivation for this change

Fixes #96417.

###### Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [x] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [x] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [x] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

Commit:
modem-manager-gui: 0.0.19.1 -> 0.0.20

Upstream source is on Sourceforge, but that's HTTP-only.

Commit:
modem-manager-gui: remove docs patch

Included in current release

worldofpeace commented on pkgs/applications/networking/modem-manager-gui/default.nix:
```diff
@@ -18,22 +18,14 @@
-    url = "https://linuxonly@bitbucket.org/linuxonly/modem-manager-gui";
-    rev = "version ${version}";
-    sha256 = "11iibh36567814h2bz41sa1072b86p1l13xyj670pwkh9k8kw8fd";
+    url = "http://hg.code.sf.net/p/modem-manager-gui/code";
```
is there no https source?

worldofpeace replied:
hmm, I guess from looking that seems to be the best place aside from a mirror

galagora replied:
Https works

worldofpeace replied:
ahh awesome. when I tried it in my browser it asked for authentication.

worldofpeace replied:
It actually doesn't
```
checking outputs of '/nix/store/iswv8scz05hnd68na1h6whxz5hinds5q-hg-archive.drv'...
getting https://hg.code.sf.net/p/modem-manager-gui/code (v0.0.20) into /nix/store/7k55mmphigcpayxwwcr6lw11ib4nb56p-hg-archive
warning: connection security to hg.code.sf.net is disabled per current settings; communication is susceptible to eavesdropping and tampering
abort: http authorization required for https://hg.code.sf.net/p/modem-manager-gui/code
error: --- Error --------------------------------------------------------------------------------------------------------------------------------------------------- nix-build
error: --- Error --- nix-daemon
builder for '/nix/store/iswv8scz05hnd68na1h6whxz5hinds5q-hg-archive.drv' failed with exit code 255; last 3 log lines:
  getting https://hg.code.sf.net/p/modem-manager-gui/code (v0.0.20) into /nix/store/7k55mmphigcpayxwwcr6lw11ib4nb56p-hg-archive
  warning: connection security to hg.code.sf.net is disabled per current settings; communication is susceptible to eavesdropping and tampering
  abort: http authorization required for https://hg.code.sf.net/p/modem-manager-gui/code
```

it probably worked for you because the archive was cached by nix. I had to do `nix-build -A modem-manager-gui.src`. Could you revert back?

galagora replied:
Wait, sorry, that was the version from the store. The authorization breaks the package.

galagora replied:
So, I could fetch it from another distro, like Debian (https://salsa.debian.org/debian/modem-manager-gui).

worldofpeace replied:
That actually isn't uncommon in nixpkgs and would be acceptable. Go for it.

worldofpeace reviewed (changes requested):
I'm seeing this uses libappindicator-gtk3 which is dead.  see https://github.com/NixOS/nixpkgs/issues/96420. I did confirm in the changelog that it supports it http://hg.code.sf.net/p/modem-manager-gui/code/file/tip/Changelog#l5

Commit:
modem-manager-gui: replace libappindicator-gtk3

libappindicator-gtk3 is unmaintained, should be replaced with fork (see #96420)

galagora commented:
OK, it runs, at least.

worldofpeace commented on pkgs/applications/networking/modem-manager-gui/default.nix:
```diff
@@ -18,22 +18,14 @@
-    url = "https://linuxonly@bitbucket.org/linuxonly/modem-manager-gui";
-    rev = "version ${version}";
-    sha256 = "11iibh36567814h2bz41sa1072b86p1l13xyj670pwkh9k8kw8fd";
+  src = fetchgit {
+    url = "https://salsa.debian.org/debian/modem-manager-gui";
+    rev = "upstream/${version}";
+    sha256 = "1pjx4rbsxa7gcs628yjkwb0zqrm5xq8pkmp0cfk4flfk1ryflmgr";
   };
```
you can use `fetchFromGitlab`
```
fetchFromGitlab {
  domain = "salsa.debian.org";
  owner =
  repo
  rev = 
  ...
```

worldofpeace commented:
About the ofono drop https://github.com/NixOS/nixpkgs/commit/0ae487ecaafb502359d7de63ab65fb0c42b89a9c#diff-d63e10da780ee77c1f16e36d5a02ce17, is that not required to build a plugin for it in this project?

galagora commented:
Yeah, oFono is just a library, I thought it was like ModemManager, which is also a daemon.

galagora commented:
Hey, @worldofpeace, this should be good, right? Your requested changes are still pending because I force pushed and can't resolve them, but I changed the lib.

worldofpeace commented:
thx @galagora :sparkles:
